packages/tldraw/src/lib/shapes/line/LineShapeUtil.tsx

```diff
@@ -45,27 +46,18 @@ export class LineShapeUtil extends ShapeUtil<TLLineShape> {
 	override hideSelectionBoundsBg = () => true
 
 	override getDefaultProps(): TLLineShape['props'] {
+		const [startIndex, endIndex] = getIndices(2)
 		return {
 			dash: 'draw',
 			size: 'm',
 			color: 'black',
 			spline: 'line',
 			handles: {
-				start: {
-					id: 'start',
-					type: 'vertex',
-					canBind: false,
-					canSnap: true,
-					index: 'a1' as IndexKey,
+				[startIndex]: {
```

Comment:
Why is it necessary to use indices here at keys? I would expect the handles to just be an array of VecModels. We use indices specifically to avoid cases where changing one record's place in order would mean changing other records, too. (If a shape's index was an integer, then moving a shape to the back of its parent's children would mean also incrementing all of its siblings' indices). That doesn't matter for records because we update "whole records", so the diff would be the same.

Reply:
just because doing all that array/index juggling is kinda fiddly tbh? fractional indexes are just a bit nicer for these types of things imo. i could change it but it would make how we update this a fair bit more complicated so i'd rather not